protobuf/endpoints/register_function.proto

```diff
@@ -4,8 +4,55 @@ package endpoints;
 
 import "models/registered_function.proto";
 
+message CpuRequirements {
+  // Optional field where 1000 millicores equals 1 CPU.
+  // From https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu:
+  // "For CPU resource units, the quantity expression 0.1 is equivalent to the expression 100m, which can be read as 'one hundred millicpu'."
+  optional uint32 millicores = 1;
+
+  // Optional field indicating amount of memory needed.
+  // Not all precisions may be accepted.
+  // For example, the worker may choose to only accept values that are multiples of Mebibytes.
+  optional uint64 memory_in_bytes = 2;
+
+  // Optional field indicating amount of storage needed.
+  // Not all precisions may be accepted.
+  // For example, the worker may choose to only accept values that are multiples of Mebibytes.
+  optional uint64 storage_in_bytes = 3;
+}
+
+message GpuRequirements {
+  // Required field indicating which GPU machine should be used to execute the function.
+  optional string gpu_model = 1;
```

Comment:
The comment says `Required`.

Reply:
Yeah, proto3 doesn't actually allow you to make a field required. Maybe it's the wrong move, but I'm trying to leave comments on most fields to indicate whether or not they are actually optional. Or in other words, I'm trying to indicate if the server will expect their presence and return an error if they're missing.